crates/core/src/client/mod.rs

```diff
@@ -112,23 +108,16 @@ impl<P: Provider + Send + Sync> KakarotEthApi<P> for KakarotClient<P> {
 
     /// Returns the result of executing a call on a ethereum address for a given calldata and block
     /// without creating a transaction.
-    async fn call_view(
-        &self,
-        ethereum_address: Address,
-        calldata: Bytes,
-        block_id: BlockId,
-    ) -> Result<Bytes, EthApiError<P::Error>> {
+    async fn call_view(&self, to: Address, calldata: Bytes, block_id: BlockId) -> Result<Bytes, EthApiError<P::Error>> {
         let starknet_block_id: StarknetBlockId = EthBlockId::new(block_id).try_into()?;
 
-        let ethereum_address: Felt252Wrapper = ethereum_address.into();
-        let ethereum_address = ethereum_address.into();
+        let to: Felt252Wrapper = to.into();
+        let to = to.into();
 
         let mut calldata = calldata.clone().into_iter().map(FieldElement::from).collect::<Vec<_>>();
 
-        let result = self
-            .kakarot_contract
-            .eth_call(&self.starknet_provider, &ethereum_address, &mut calldata, &starknet_block_id)
-            .await?;
+        let result =
+            self.kakarot_contract.eth_call(&self.starknet_provider, &to, &mut calldata, &starknet_block_id).await?;
```

Comment:
why do we `mut` the `calldata` btw?

Reply:
My mistake, I though you had to make calldata mutable in order to append to it in the `eth_call` function, by turns out you can just mark it as mut in the `eth_call` function signature.